crawler/file_processing.py

```diff
@@ -454,10 +545,11 @@ def get_db(self) -> Database:
         Returns:
             Database -- a reference to the database in mongo
         """
-        client = create_mongo_client(self.config)
-        db = get_mongo_db(self.config, client)
+        if not hasattr(self, "db"):
+            client = create_mongo_client(self.config)
+            self.db = get_mongo_db(self.config, client)
 
-        return db
+        return self.db
```

Comment:
does it need to be self? is that for the tests?

Reply:
this is to always return the same database connector, and not create a new one in every call